### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by nikkypyra**

### CODE REVIEW

File: src/components/AuthContext.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    await db
+      .collection('users')
+      .doc(auth.currentUser.uid)
+      .get()
+      .then((doc) => {
+        return doc.exists && doc.data()
+      })
+      .then(async (data) => {
+        await setProfile({ ...data })
+      })
```
Comment (carolinstuhr) [inline src/components/AuthContext.js:43]: I might be wrong but couldn't you leave out the async and await here as you are already using .then which is a promise? so just write .then((data) => setProfile({...data}) ?

File: src/pages/Login.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            'The email or password you entered did not match our records. Please double-check and try again.'
+          )
+        }
+        if (res.code === 'auth/wrong-password' && loginCounter <= 3) {
+          setLoginCounter(loginCounter + 1)
+          return setError(
+            'password',
+            'reset',
+            'The email or password you entered did not match our records. You may try again or  '
+          )
```
Comment (carolinstuhr) [inline src/pages/Login.js:85]: Isn't there something missing after the or?

**Review by carolinstuhr — COMMENTED** [review]
Very impressive work, I only made two little notes.

File: src/components/AuthContext.js (hunk 1 above)
Comment (nikkypyra) [reply src/components/AuthContext.js:43]: You could be right, I will give that a try.

File: src/pages/Login.js (hunk 2 above)
Comment (nikkypyra) [reply src/pages/Login.js:85]: the rest of the message is separated as a link to reset the password :)

**Pull request merged**